clients/src/main/java/org/apache/kafka/clients/admin/DescribeTopicsOptions.java

```diff
@@ -47,8 +49,32 @@ public DescribeTopicsOptions includeAuthorizedOperations(boolean includeAuthoriz
         return this;
     }
 
+    /**
+     * Whether to use the DescribeTopicPartitions API. It should be set to false if DescribeTopicPartitions API is
+     * not supported.
+     *
+     */
+    public DescribeTopicsOptions useDescribeTopicPartitionsApi(boolean useDescribeTopicPartitionsApi) {
```

Comment:
I think you need a small change to KIP-966 to document these changes to the admin API.

Reply:
Shouldn't the selection be automatic? I don't think uses will bother about this. Basically, the new API should be used when available and the old one when not.

Reply:
Removed useDescribeTopicPartitionsApi. It was only used in the UnsupportedVersionException retry, but now I figured out a way to retry in the Call framework.